scene/resources/capsule_shape_2d.cpp

```diff
@@ -32,6 +32,25 @@
 #include "servers/physics_2d_server.h"
 #include "servers/visual_server.h"
 
+Vector<Vector2> CapsuleShape2D::_get_points() const {
+
+    Vector<Vector2> points;
+	for (int i = 0; i < 24; i++) {
+		Vector2 ofs = Vector2(0, (i > 6 && i <= 18) ? -get_height() * 0.5 : get_height() * 0.5);
+
+		points.push_back(Vector2(Math::sin(i * Math_PI * 2 / 24.0), Math::cos(i * Math_PI * 2 / 24.0)) * get_radius() + ofs);
+		if (i == 6 || i == 18)
+			points.push_back(Vector2(Math::sin(i * Math_PI * 2 / 24.0), Math::cos(i * Math_PI * 2 / 24.0)) * get_radius() - ofs);
+	}
+
+	return points;
+}
+
+bool CapsuleShape2D::_edit_is_clicked(const Point2 &p_point, double p_tolerance) const {
+
+	return Geometry::is_point_in_polygon(p_point, _get_points());
```

Comment:
Wut ? Calculating whether a point is in a capsule or not does not require you to create a polygon...
Just check whether it is in the "body rectangle" then in the two extermity circles.

Reply:
I know, but why spend time optimizing this and introducing just another possibility of mismatch between drawing and selection code.

Reply:
Well, you're building a polygon to check for 2 circles and a square, this is not optimizing. 
Honestly, if the drawing and the selection code mismatch a little bit it is not important. The code here is overly complex for little gain. Just keep thing simple, in this case maintainability is probably more important than accuracy. :)